test/integration/plugins/nuagevsp/nuageTestCase.py

```diff
@@ -77,85 +117,104 @@ def setUpClass(cls, zone=None):
+            providers = filter(lambda physical_network:
+                               "VSP" in physical_network.isolationmethods,
+                               pns)[0].providers
+            devices = filter(lambda provider:
+                             provider.name == "NuageVsp",
+                             providers)[0].devices
             cls.nuage_vsp_device.username = devices[0].username
             cls.nuage_vsp_device.password = devices[0].password
             cls.cms_id = cls.nuage_vsp_device.cmsid
         except Exception as e:
             cls.tearDownClass()
-            raise unittest.SkipTest("Warning: Could not get configured Nuage VSP device details - %s" % e)
+            raise unittest.SkipTest("Warning: Could not get configured "
+                                    "Nuage VSP device details - %s" % e)
+        return
 
-        # VSD is a programmable policy and analytics engine of Nuage VSP SDN platform
-        # vspk is a Python SDK for Nuage VSP's VSD
-        # libVSD is a library that wraps vspk package
+    @classmethod
+    def configureVSDSessions(cls):
+        # VSD is a programmable policy and analytics engine of Nuage VSP SDN
+        # platform; vspk is a Python SDK for Nuage VSP's VSD; libVSD is a
+        # library that wraps vspk package
         try:
-            vspk_module = "vspk." + cls.nuage_vsp_device.apiversion if int(cls.nuage_vsp_device.apiversion[1]) >= 4 \
+            vspk_module = "vspk." + cls.nuage_vsp_device.apiversion \
+                if int(cls.nuage_vsp_device.apiversion[1]) >= 4 \
                 else "vspk.vsdk." + cls.nuage_vsp_device.apiversion
             cls.vsdk = importlib.import_module(vspk_module)
             from libVSD import ApiClient, VSDHelpers
         except Exception as e:
             cls.tearDownClass()
-            raise unittest.SkipTest("Warning: vspk (and/or) libVSD package import failure - %s" % e)
+            raise unittest.SkipTest("Warning: vspk (and/or) libVSD package "
+                                    "import failure - %s" % e)
 
         # Configure VSD session
-        cls._session = cls.vsdk.NUVSDSession(username=cls.nuage_vsp_device.username,
-                                             password=cls.nuage_vsp_device.password,
-                                             enterprise="csp",
-                                             api_url="https://%s:%d" % (cls.nuage_vsp_device.hostname,
-                                                                        cls.nuage_vsp_device.port)
-                                             )
+        cls._session = cls.vsdk.NUVSDSession(
+            username=cls.nuage_vsp_device.username,
+            password=cls.nuage_vsp_device.password,
+            enterprise="csp",
+            api_url="https://%s:%d" % (cls.nuage_vsp_device.hostname,
+                                       cls.nuage_vsp_device.port)
+            )
```

Comment:
Please add an assert to validate that ``cls._session`` is valid.  Otherwise, an invalid session may cause subsequent tests to fail by side effect inhibiting debugging.